reactor-core/src/main/java/reactor/core/publisher/FluxDistinctUntilChanged.java

```diff
@@ -210,8 +220,7 @@ public boolean tryOnNext(@Nullable T t) {
 				return true;
 			}
 
-			if (Objects.equals(lastKey, k)) {
-				lastKey = k;
+			if (null != lastKey && keyComparator.test(lastKey, k)) {
```

Comment:
`lastKey` must still be initialized to `k`. This block must also be protected from the comparator throwing an exception (like what is done just above).